tests/test_contracts.py

```diff
@@ -24,6 +25,15 @@ class ResponseMock(object):
     url = 'http://scrapy.org'
 
 
+class CustomFormContract(Contract):
+    name = 'custom_form'
+
+    def adjust_request_args(self, args):
+        args['request_cls'] = FormRequest
```

Comment:
I was thinking about doing it in a different way:

```py
class CustomFormContract(Contract):
    name = 'custom_form'
    request_cls = FormRequest

    def adjust_request_args(self, args):
         args['formdata'] = {'name': 'scrapy'}
         return args
```
Not passing request_cls in `args` looks cleaner to me, because `request_cls` is not really a keyword argument passed to a request initializer.

That said, if it is not an request arg, we'd need some rule to determine which class to use if there are several contracts in a chain. Something like "take last explicitly defined request_cls".